file-formats/nrob/nrob.json

```diff
@@ -0,0 +1,157 @@
+ng and the lowest number is assigned again. You can suppress this behavior by setting this option false. If you do this, no more characters from an interval are assigned once the characters of the interval are used up. You can suppress rollover for t
+he entire number range object, that is, the setting applies to all the intervals contained in the object.",
+          "type": "boolean"
+        },
+        "prefix": {
+          "title": "Prefix",
+          "description": "If it is set to true determined numbers consist of the prefix (name of subobject) and the numbers.",
+          "type": "boolean"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "numberLengthDomain",
+        "percentage",
+        "subType",
+        "untilYear",
+        "rolling",
+        "prefix"
+      ]
+    },
+    "configuration": {
+      "title": "Configuration",
+      "description": "Configuration-specific fields",
+      "type": "object",
+      "properties": {
+        "transactionId": {
+          "title": "Transaction ID",
+          "description": "Transaction code for application specific transaction",
+          "type": "string",
+          "maxLength": 20
+        },
+        "buffering": {
+          "title": "Buffering",
+          "description": "Choose a buffer type for no buffering, for buffering via main memory or for parallel buffering. Default is mainBuffer.",
+          "type": "string",
+          "enum": [
+            "mainBuffer",
+            "parallel",
+            "none"
+          ],
+          "enumDescriptions": [
+            "Main memory buffering. \n\n Main memory buffering is the most efficient buffering solution with regards to system performance. Almost all number range objects use main memory buffering. However, it does have two restrictions: Numbers can
+ no longer be assigned in chronological order. The numbering may contain gaps.",
+            "Parallel buffering, \n\n Parallel buffering works in the same way as main memory buffering, with the difference that the number packages are not stored in the main memory, but are written to the table NRIVSHADOW.",
+            "No buffering \n\n A number is taken from an interval stored in database NRIV. The table line is locked and this lock remains until the application that requested the number is closed, either with a COMMIT WORK or a ROLLBACK."
+          ]
+        },
+        "bufferedNumbers": {
+          "title": "Buffered Numbers",
+          "description": "This value specifies the numbers in buffer. In case of parallel and main memory buffering, add a number for \u0027bufferedNumbers\u0027. It determines how many numbers are reserved in buffer for the intervals.Default number
+of buffers is 10.",
+          "type": "string",
+          "maxLength": 8
```

Comment:
bufferedNumbers in backend is numc(8). A positive number with 8 digits.
How can we replace numc(8)?

Reply:
What do you mean with replace?

Reply:
The original type of buffered_numbers was data element nrivbuffer, which is base type (domaine) num8 (data type numc(8)).
But we cannot use num8 as type of buffered_numbers  in the interface, because then the type is not self contained in the interface.

Reply:
Couldn't we just add support for numeric types by adding a regular expression that allows only values between 0-9. See https://json-schema.org/understanding-json-schema/reference/string.html#regular-expressions

Reply:
Done. We added a "pattern" to the schema generator when the type is "n".